### PR TITLE
Bulk delete

### PR DESCRIPTION
**Pull request by VagantemNumen**
Added bulk delete endpoint and bulk delete support.

### CODE REVIEW

File: restapi.go [hunk 1]
```diff
@@ -1107,6 +1107,34 @@ func (s *Session) ChannelMessageDelete(channelID, messageID string) (err error)
...
+		return
+	}
+
+	if len(messages) == 1 {
+		err = s.ChannelMessageDelete(channelID, messages[0])
+		return
+	}
+
+	if len(messages) > 100 {
+		messages = messages[:100]
```
Comment (iopred) [inline restapi.go]: Perhaps you could call bulk delete multiple times for each group of 100 messages?

**Comment by VagantemNumen** [comment]
This is why I did not do it like that,
```
warn This endpoint has a rate limit of 1 request per second per guild, and is limited to 
100 messages per request. 
The request will fail if less than two snowflakes are provided. 
Only bot accounts can use this endpoint.
```

File: restapi.go (hunk 1 above)
Comment (VagantemNumen) [inline restapi.go]: This is why I did not do it like that,
```
warn This endpoint has a rate limit of 1 request per second per guild, and is limited to 
100 messages per request. 
The request will fail if less than two snowflakes are provided. 
Only bot accounts can use this endpoint.
```

Comment (iopred) [inline restapi.go]: Nice, Looks good to me!

**Comment by bwmarrin** [comment]
Great job @VagantemNumen and thanks a lot for the help! I really appreciate it!

**Pull request merged**